mne/viz/epochs.py

```diff
@@ -93,319 +94,376 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
-    ylims = {ch_type: (1., -1.) for ch_type in all_ch_types}
-    for (epochs_, ch_type, ax_name, name, data, overlay_times,
-         vmin, vmax, ts_args) in groups:
-        vmin, vmax = vmins.get(ch_type, vmin), vmaxs.get(ch_type, vmax)
-        these_axes = axes[ax_name] if isinstance(axes, dict) else axes
-        axes_dict = _prepare_epochs_image_axes(these_axes, fig, colorbar,
-                                               evoked)
-        axes_list.append(axes_dict)
-        title_ = ((ax_name if isinstance(axes, dict) else name)
-                  if title is None else title)
+    for this_group, this_group_dict in group_by.items():
+        this_norm = this_group_dict['norm']
+        this_image = this_group_dict['image']
+        this_epochs = this_group_dict['epochs']
+        this_ch_type = this_group_dict['ch_type']
+        this_axes_dict = this_group_dict['axes']
+        this_title = this_group_dict['title']
+        vmin, vmax = vmin_vmax[this_ch_type]
+
+        # plot title
+        if this_title is None:
+            title = _handle_default('titles').get(this_group, this_group)
+            if isinstance(combine, str) and len(title):
+                _comb = combine.upper() if combine == 'gfp' else combine
+                _comb = 'std. dev.' if _comb == 'std' else _comb
+                title += ' ({})'.format(_comb)
+
+        # plot the image
         this_fig = _plot_epochs_image(
-            data, times, style_axes=True, epochs=epochs_, vmin=vmin, vmax=vmax,
-            colorbar=colorbar, show=False, unit=units[ch_type], cmap=cmap,
-            ax=axes_dict, title=title_, overlay_times=overlay_times,
-            evoked=evoked, ts_args=ts_args)
-        figs.append(this_fig)
-
-        # the rest of the code is for aligning ylims for multiple plots
-        if evoked is True and not manual_ylims:
-            evoked_ax = axes_dict["evoked"]
-            this_min, this_max = evoked_ax.get_ylim()
-            curr_min, curr_max = ylims[ch_type]
-            ylims[ch_type] = min(curr_min, this_min), max(curr_max, this_max)
-
-    if evoked is True:  # adjust ylims
-        for group, axes_dict in zip(groups, axes_list):
-            ch_type = group[1]
-            ax = axes_dict["evoked"]
-            this_ymin, this_ymax = these_ylims = ylims[ch_type]
-            ax.set_ylim(these_ylims)
-            yticks = np.array(ax.get_yticks())
-            max_height = yticks[yticks < this_ymax][-1]
+            this_image, style_axes=True, epochs=this_epochs, picks=picks,
+            colorbar=colorbar, vmin=vmin, vmax=vmax, norm=this_norm, cmap=cmap,
+            unit=units[this_ch_type], ax=this_axes_dict, show=False,
+            title=title, combine=combine, combine_given=combine_given,
+            overlay_times=overlay_times, evoked=evoked, ts_args=ts_args)
+        group_by[this_group].update(fig=this_fig)
+
+        # detect ylims across figures
+        if evoked and not manual_ylims:
```

Comment:
I don't like `evoked`. I guess I named it thus originally, but should it not be `do_evoked`? `evoked` is usually an evoked object. it doesn't sound like a flag.

Reply:
And shouldn't this be in 344 and just be `if not manual_ylims`?

Reply:
changing to `do_evoked` would be another API change, since this variable comes from the call signature. If you're OK with that, I would vote for `show_evoked` rather than `do_evoked`.

This block actually can't be moved down to 344 though; we need to cycle through the figures twice (once to detect all the ylims, and a second time to set them) so it needs to be in this earlier for-loop.